CRM/Member/Form/Membership.php

```diff
@@ -638,26 +638,8 @@ public function buildQuickForm() {
         )
       );
 
-      $allowStatuses = array();
-      $statuses = CRM_Contribute_PseudoConstant::contributionStatus();
-      if ($this->_onlinePendingContributionId) {
-        $statusNames = CRM_Contribute_PseudoConstant::contributionStatus(NULL, 'name');
-        foreach ($statusNames as $val => $name) {
-          if (in_array($name, array(
-            'In Progress',
-            'Overdue',
-          ))
-          ) {
-            continue;
-          }
-          $allowStatuses[$val] = $statuses[$val];
-        }
-      }
-      else {
-        $allowStatuses = $statuses;
-      }
       $this->add('select', 'contribution_status_id',
-        ts('Payment Status'), $allowStatuses
+        ts('Payment Status'), CRM_Contribute_BAO_Contribution_Utils::getContributionStatuses('membership')
```

Comment:
@seamuslee001 I don't think there is any requirement of those statuses irrespective of whether the membership has ```$_ _onlinePendingContributionId``` or not.  So we should simply exclude those two statuses for membership payment. What you say? These statuses are later excluded [here](https://github.com/civicrm/civicrm-core/pull/10718/files#diff-99d052aa6a966c5a2cc89d05f057d22dR545)